go/runtime/host/tdx/qemu.go

```diff
@@ -211,6 +232,47 @@ func (q *qemuProvisioner) getSandboxConfig(rtCfg host.Config, _ sandbox.Connecto
 	return cfg, nil
 }
 
+// createPersistentOverlayImage creates a persistent overlay image for the given backing image and
+// returns the full path to the overlay image. In case the image already exists, it is reused.
+//
+// The format of the resulting image is always qcow2.
+func (q *qemuProvisioner) createPersistentOverlayImage(
+	rtCfg host.Config,
+	comp *bundle.ExplodedComponent,
+	image string,
+	format string,
+) (string, error) {
+	compID, _ := comp.ID().MarshalText()
+	imageDir := filepath.Join(q.cfg.DataDir, persistentImageDir, rtCfg.ID.String(), string(compID))
+	imageFn := filepath.Join(imageDir, fmt.Sprintf("%s.overlay", filepath.Base(image)))
+	if _, err := os.Stat(imageFn); err == nil {
```

Comment:
```suggestion
	if _, err := os.Stat(imageFn); err == nil { // err is nil
```
Add some comment to notify the reader that this is non-idiomatic check, as one expects `err != nil`.

You should thrown an error if `Stat` failed and create a file when doesn't exist.

Reply:
Fixed.